config/lib/express.js

```diff
@@ -40,8 +40,9 @@ module.exports.initLocalVariables = function (app) {
 
   // Passing the request url to environment locals
   app.use(function (req, res, next) {
-    res.locals.host = req.protocol + '://' + req.hostname;
-    res.locals.url = req.protocol + '://' + req.headers.host + req.originalUrl;
+    var domain = config.app.domain || req.headers.hostname;
```

Comment:
@Koblaid I meant that we could use `req.hostname` when there is no domain set in the env config; as a default instead of `req.headers.hostname`. This way a user of this project could implement the domain, if they are concerned with this vulnerability; for instance, when the project is early dev or used internally/locally only. Otherwise, if the domain is omitted then it would default to it's current behavior.